subsys/bluetooth/host/gatt.c

```diff
@@ -274,6 +275,29 @@ static void sc_process(struct k_work *work)
 	atomic_set_bit(sc->flags, SC_INDICATE_PENDING);
 }
 
+#if defined(CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE)
+static struct gatt_ccc_store {
+	struct bt_conn *conn;
+	atomic_t scheduled;
+	struct k_delayed_work work;
+} gatt_ccc_store;
+
+static void ccc_delayed_store(struct k_work *work)
+{
+	struct gatt_ccc_store *ccc_store =
+		CONTAINER_OF(work, struct gatt_ccc_store, work);
+
+	if (bt_addr_le_is_bonded(ccc_store->conn->id,
+		&ccc_store->conn->le.dst)) {
+		bt_gatt_store_ccc(ccc_store->conn->id,
+			&ccc_store->conn->le.dst);
+	}
+
+	bt_conn_unref(conn);
```

Comment:
Did you even compile test this? I don't see any `conn` variable in this function. Also, if/when you're unrefing some shared pointer you should always set it to NULL afterward.